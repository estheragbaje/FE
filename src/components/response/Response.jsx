import React, { Component } from 'react';
import { withApollo } from 'react-apollo';
import Divider from '../../styles/Divider';
import { GET_SURVEY_FEEDBACK } from '../../graphql/queries';
import ResultSort from './ResultSort';
import DashboardLayout from '../dashboard/layouts/DashboardLayout';


class Response extends Component {
  constructor(props) {
    super(props);
    this.state = {
      surveyId: '',
      survey: null,
      loading: false,
    };
  }

  componentDidMount() {
    const {
      match: {
        params: { surveyId },
      },
    } = this.props;
    this.setState({ surveyId });

    this.getSurveyDetails(surveyId);
  }

  getSurveyDetails= async (surveyId) => {
    const {
      client,
    } = this.props;
    // console.log(client);
    this.setState({ loading: true });
    const { data, loading } = await client.query({
      query: GET_SURVEY_FEEDBACK, variables: { surveyId },
    });
    if (!loading && data) {
      const survey = data.getSurveyFeedback;
      this.setState({ survey, loading });
    }
    // console.log(data);
  }

  render() {
    const { survey } = this.state;
    return (
      <DashboardLayout>
        <h1 className="text-center create-survey-title f-1">
          {survey && survey.title}
        </h1>
        <Divider size={30} />
        {survey && survey.questions && <ResultSort questions={survey.questions} />}
      </DashboardLayout>
    );
  }
}


export default withApollo(Response);
