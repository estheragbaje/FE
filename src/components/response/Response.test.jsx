import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Response } from './Response';

const mockProps = {
  match: {
    params: { surveyId: 'c71ca084-6c21-4ef8-b4fe-566ebb06c994' },
  },
  client: {
    query: jest.fn(),
    mutate: jest.fn(),
  },
};

const mockSurvey = {
  title: 'test title',
  id: '234455-2345',
  questions: [
    {
      id: 1,
      type: 'text',
      question: 'What do you say about my overall usefulness in the team for the past 4 months?',
      feedbacks: [
        {
          id: 1,
          rating: '7',
          comment: null,
        },
        {
          id: 2,
          rating: null,
          comment: "It's been quite incredible",
        },
      ],
    },
    {
      id: 2,
      type: 'text',
      question: 'What do you say about my overall usefulness in the team for the past 4 months?',
      feedbacks: [
        {
          id: 1,
          rating: null,
          comment: "It's been quite incredible",
        },
        {
          id: 2,
          rating: 3,
          comment: null,
        },
      ],
    },
  ],
};

let wrapper;
beforeAll(() => {
  // eslint-disable-next-line react/jsx-props-no-spreading
  wrapper = shallow(<Response {...mockProps} />);
});
describe('<Response />', () => {
  it('renders the Response component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should mount the component', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getSurveyDetails');
    instance.componentDidMount();
    expect(instance.getSurveyDetails).toHaveBeenCalledWith(
      mockProps.match.params.surveyId,
    );
    expect(instance.state.surveyId).toEqual(mockProps.match.params.surveyId);
  });

  it('should initialize state if surveyId params is defined', () => {
    const instance = wrapper.instance();

    jest.spyOn(instance, 'getSurveyDetails').mockImplementation(
      jest.fn(() => {
        instance.setState({ loading: false, survey: mockSurvey });
      }),
    );
    instance.componentDidMount();
    expect(instance.state.survey).toEqual(mockSurvey);
  });
});
