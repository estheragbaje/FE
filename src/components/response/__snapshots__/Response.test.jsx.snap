// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Response /> renders the Response component correctly 1`] = `
<DashboardLayout>
  <Spinner />
  <h1
    className="text-center create-survey-title f-1"
  />
  <styled.div
    size={30}
  />
</DashboardLayout>
`;
